WordPress/Classes/Services/MediaLibrary.swift

```diff
@@ -1,7 +1,192 @@
+                    }
+                }
+            })
+        }
+    }
+
+    /// Creates a Media object with a UIImage, asynchronously.
+    ///
+    /// The UIImage is expected to be a JPEG, PNG, or other 'normal' image.
+    ///
+    /// - parameter onMedia: Called if the Media was successfully created and the image's data exported to an absoluteLocalURL.
+    /// - parameter onError: Called if an error was encountered during creation, error convertible to NSError with a localized description.
+    ///
+    public func makeMediaWith(blog: Blog, image: UIImage, onMedia: @escaping (Media) -> (), onError: ((Error) -> ())?) {
+        DispatchQueue.global(qos: .default).async {
+
+            let exporter = MediaImageExporter()
+            exporter.maximumImageSize = self.exporterMaximumImageSize()
+            exporter.stripsGeoLocationIfNeeded = MediaSettings().removeLocationSetting
+
+            exporter.exportImage(image, fileName: nil, onCompletion: { (imageExport) in
+                self.managedObjectContext.perform {
+
+                    let media = Media.makeMedia(blog: blog)
+                    self.configureMedia(media, withExport: imageExport)
+                    onMedia(media)
+                }
+            }, onError: { (error) in
+                if let onError = onError {
+                    self.managedObjectContext.perform {
+
+                        let nerror = error.toNSError()
+                        DDLogSwift.logError("Error occurred exporting Media with a UIImage, code: \(nerror.code), error: \(nerror)")
+                        onError(error.toNSError())
+                    }
+                }
+            })
+        }
+    }
+
+    /// Creates a Media object with a file at a URL, asynchronously.
+    ///
+    /// The file URL is expected to be a JPEG, PNG, GIF, other 'normal' image, or video.
+    ///
+    /// - parameter onMedia: Called if the Media was successfully created and the file's data exported to an absoluteLocalURL.
+    /// - parameter onError: Called if an error was encountered during creation, error convertible to NSError with a localized description.
+    ///
+    public func makeMediaWith(blog: Blog, url: URL, onMedia: @escaping (Media) -> (), onError: ((Error) -> ())?) {
+        DispatchQueue.global(qos: .default).async {
+            let exporter = MediaURLExporter()
+
+            exporter.maximumImageSize = self.exporterMaximumImageSize()
+            exporter.stripsGeoLocationIfNeeded = MediaSettings().removeLocationSetting
+
+            exporter.exportURL(fileURL: url, onCompletion: { (urlExport) in
+                self.managedObjectContext.perform {
+
+                    let media = Media.makeMedia(blog: blog)
+                    self.configureMedia(media, withExport: urlExport)
+                    onMedia(media)
+                }
+            }, onError: { (error) in
+                if let onError = onError {
+                    self.managedObjectContext.perform {
```

Comment:
These error handlers all do the same, I think... could we define a single `handleError` method that we just pass to `onError`? (`onError: self.handleError(error)`)